server/src/apub/post.rs

```diff
@@ -51,6 +51,46 @@ impl ToApub for Post {
     let url = self.url.as_ref().filter(|u| !u.is_empty());
     if let Some(u) = url {
       oprops.set_url_xsd_any_uri(u.to_owned())?;
+
+      // Embeds
+      let mut page_preview = Page::new();
+      page_preview
+        .object_props
+        .set_url_xsd_any_uri(u.to_owned())?;
+
+      if let Some(embed_title) = &self.embed_title {
+        page_preview
+          .object_props
+          .set_name_xsd_string(embed_title.to_owned())?;
+      }
+
+      if let Some(embed_description) = &self.embed_description {
+        page_preview
+          .object_props
+          .set_summary_xsd_string(embed_description.to_owned())?;
+      }
+
+      if let Some(embed_html) = &self.embed_html {
+        page_preview
+          .object_props
+          .set_content_xsd_string(embed_html.to_owned())?;
+      }
+
+      oprops.set_preview_base_box(page_preview)?;
```

Comment:
Is it correct that it uses a Page as preview for a Page? Seems a bit strange but I dont know how its supposed to work, the activitystreams spec only has an example with a link object as preview.

https://www.w3.org/TR/activitystreams-core/#example-3

Reply:
Since its a preview of a web page / url, it made sense to do it as a page. I'm not exactly sure the difference between a page and a link.